epochStart/mock/runTypeComponentsStub.go

```diff
@@ -20,6 +23,22 @@ func NewRunTypeComponentsStub() *RunTypeComponentsStub {
 		AdditionalStorageServiceFactory:  &testscommon.AdditionalStorageServiceFactoryMock{},
 		ShardCoordinatorFactory:          sharding.NewMultiShardCoordinatorFactory(),
 		NodesCoordinatorWithRaterFactory: nodesCoordinator.NewIndexHashedNodesCoordinatorWithRaterFactory(),
+		RequestHandlerFactory:            requestHandlers.NewResolverRequestHandlerFactory(),
+	}
+}
+
+// NewSovereignRunTypeComponentsStub -
+func NewSovereignRunTypeComponentsStub() *RunTypeComponentsStub {
+	runTypeComponents := NewRunTypeComponentsStub()
+
+	additionalStorageServiceCreator, _ := storageFactory.NewSovereignAdditionalStorageServiceFactory()
+	requestHandlerFactory, _ := requestHandlers.NewSovereignResolverRequestHandlerFactory(runTypeComponents.RequestHandlerFactory)
```

Comment:
Same remark, this is not a stub anymore...

Reply:
it's not... but it will now work later if we will create tests for sovereign code coverage...

Reply:
When we'll create those type of tests, we can directly replace the needed mocks with real components. If tests work without real components, I would suggest we only use mocks for now

Reply:
changed to mocks where it was possible